### PR TITLE
stm32 xspi driver supports Memory mapped mode for XiP

### PR DESCRIPTION
**Pull request by FRASTM**
Like the flash_stm32_qspi or flash_stm32_ospi drivers, this PR adds the Memory Mapped mode
so that code can eXecute In Place from the external  NOR flash 
This is validated on the stm32H5 serie and demonstrated on the stm32h573i_dk disco kit (Macronix MX25LM51245 octo-NOR ).

Running the samples/application_development/code_relocation_nocopy/ with CONFIG_STM32_MEMMAP=y

```
*** Booting Zephyr OS build v3.6.0-5497-g1db55c1f8901 ***                       
Address of main function 0x80005e1                                              
Address of function_in_ext_flash 0x90000001                                     
Address of var_ext_sram_data 0x200000a0 (10)                                    
Address of function_in_sram 0x20000001                                          
Address of var_sram_data 0x200000a4 (10)                                        
Hello World! stm32h573i_dk
```                                                      

Requires https://github.com/zephyrproject-rtos/zephyr/pull/73220

### CODE REVIEW

**Comment by FRASTM** [comment]
CI failure : west build -p auto -b nrf54h20dk/nrf54h20/cpuapp samples/drivers/spi_flash due to https://github.com/zephyrproject-rtos/zephyr/issues/73005

**Comment by FRASTM** [comment]
rebase on df788d0b41300473f047a1497fae57e8ab351f26

File: drivers/flash/flash_stm32_xspi.c [hunk 1]
```diff
@@ -1196,7 +1389,9 @@ static int flash_stm32_xspi_write(const struct device *dev, off_t addr,
 			break;
 		}
 	}
+	goto write_end;
```
Comment (erwango) [inline drivers/flash/flash_stm32_xspi.c:1392]: goto not required here.

Comment (erwango) [inline drivers/flash/flash_stm32_xspi.c:1392]: goto not required here, but required after stm32_xspi_mem_ready.

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: drivers/flash/flash_stm32_xspi.c (hunk 1 above)
Comment (FRASTM) [reply drivers/flash/flash_stm32_xspi.c:1392]: agree, 
moved to line 1347
```
	if (ret != 0) {
		LOG_ERR("XSPI: write not ready");
		goto write_end;
	}
```
that was just keeping the same structure as flash_stm32_ospi driver

**Review by erwango — DISMISSED** [review]

**Comment by FRASTM** [comment]
adding the configuration to execute the samples/drivers/spi_flash/ on the stm32h573i_dk with the external NOR in **MemoryMapped** mode

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**